[Ong Li Hong] iP

lihongong opened the pull request:
# Milo Ice
> "I love deadlines. I like the whooshing sound they make as they fly by." — Douglas Adams

Milo Ice – Effortless Note-Taking, One Sip at a Time. It is,
- *intuitive* 🧠
- text-based 
- ~~FAST~~ SUPER FAST & Minimalist ⚡

All you need to do is,
1. download it from [here](https://github.com/wswddl/ip/releases/download/A-Release/milo_ice.jar).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE!** 

Features:
- [x] Managing tasks
- [x] Managing deadlines
- [x] Managing events
- [ ] Reminders (coming soon)
If you are a Java programmer, this is how the `main` method looks like:
##
```java
public static void main(String[] args) {
        Application.launch(Main.class, args);
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

lihongong committed:
add MiloIce.java

lihongong committed:
add MiloIce.java

lihongong committed:
add MiloIce.java

lihongong committed:
add MiloIce.java Task.java

lihongong committed:
add MiloIce.java Task.java Deadline.java Event.java Todo.java

lihongong committed:
add MiloIce.java Task.java EXPECTED.TXT input.txt runtest.bat MiloIceException.java

lihongong committed:
add MiloIce.java Task.java

lihongong committed:
add MiloIce.java input.txt

lihongong committed:
add MiloIce.java EXPECTED.TXT Command.java

lihongong committed:
add MiloIce.java

lihongong committed:
Update MiloIce.java
Add Command.java

Use Enums in MiloIce.java for input command control.

lihongong committed:
Add Storage.java Update MiloIce.java Task.java Todo.java Deadline.java
Event.java

Storage class helps read and write task into a save text file.

lihongong committed:
Add Storage.java Update MiloIce.java

Storage class is used to handle reading and writing tasks into the saved
text file.

lihongong committed:
Merge branch 'branch-Level-7'

lihongong committed:
Update Todo.java Deadline.java Event.java MiloIce.java Storage.java

Allow date in the format of String (E.g. tonight, tomorrow, Sunday).

Invalid or meaningless date should not be allowed.

Only allow input date in the format of yyyy-MM-dd
using java.util.time.

lihongong committed:
Merge branch 'branch-Level-8'

lihongong committed:
Add AddCommand.java DeleteCommand.java ListCommand.java ExitCommand.java
ToggleMarkCommand.java Parser.java TaskList.java Ui.java Enum.java

Code lacks OOP style.

Transform the code to suit OOP style.

Parser class helps parse and understand the user input, command classes
helps differentiate between different type of input.

lihongong committed:
Merge branch 'branch-A-MoreOOP'

lihongong committed:
Package files into structured directories

Moved all files from /src/main/java/ into:
- milo
- milo.task
- milo.command

Update package declarations.

lihongong committed:
Merge branch 'branch-A-Packages'

lihongong committed:
Merge branch 'add-gradle-support'

lihongong committed:
Add JUnit tests for task classes

Help automate the testing process of classes.

lihongong committed:
Merge branch 'branch-A-JUnit'

lihongong committed:
Update build.gradle

Package the App as a JAR file.

lihongong committed:
Update the javadoc

Write the description of non-trivial public methods and classes.

lihongong committed:
Update the javadoc for the rest of the java files

Write the description of non-trivial public methods and classes.

lihongong committed:
Comply with coding standard

lihongong committed:
Add FindCommand.java

Enable finding task through keyword of the task's description.

Iterate through each task in TaskList and check whether the description
matches the keyword.

lihongong committed:
Merge branch 'branch-A-JavaDoc'

lihongong committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/milo/MiloIce.java
#	src/main/java/milo/Ui.java
#	src/main/java/milo/task/Event.java

lihongong committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/milo/Parser.java
#	src/main/java/milo/Ui.java
#	src/main/java/milo/command/DeleteCommand.java
#	src/main/java/milo/task/TaskList.java

lihongong committed:
Update javadoc and fix bugs

lihongong committed:
Implement JUnit test for classes under milo.task package

lihongong committed:
Fix bugs and implement time into Deadline and Event

quantin96 commented on src/main/java/milo/task/TaskList.java:
```diff
@@ -0,0 +1,87 @@
+ * from the list, as well as to check the size of the list.
+ */
+public class TaskList {
+    private final ArrayList<Task> listOfTask;
```
Not sure if this follows the code naming guidelines. Maybe `tasks` could be more appropriate?

lihongong replied:
Thanks for the review, I use the name `listOfTask` to show that it is of type `ArrayList`.

quantin96 reviewed (commented):
LGTM. Just a few nits to fix.

Huan-Kiat commented on src/main/java/milo/task/Event.java:
```diff
@@ -0,0 +1,104 @@
+     * @return a new Event task
+     * @throws MiloIceException if the date format is invalid
+     */
+    public static Event of(String description, boolean isDone,
```
The use of line break here is good.

lihongong replied:
Thanks for the feedback :)

Huan-Kiat commented on src/main/java/milo/Ui.java:
Perhaps there is a need to follow the format of other files and leave a line between comments and methods?

Huan-Kiat reviewed (commented):
No coding standard related issues found
LGTM

gandwarf commented:
Roughly check and no coding standard related issues found at present

gandwarf commented on src/main/java/milo/Parser.java:
```diff
@@ -0,0 +1,145 @@
+        String[] parts = input.split(" ");
+        Enum inputEnum = Enum.of(input);
+
+        if (inputEnum == Enum.BYE) {
```
Maybe this can use switch-case type to simplify the code.

lihongong committed:
Update MiloIce.java Parser.java to comply with CheckStyle standards

lihongong committed:
Implement Javafx features and add resources

lihongong committed:
Implement javafx features and change user and bot's profile picture

lihongong committed:
Add background image and make circular profile picture

lihongong committed:
Merge branch 'branch-Level-10'

lihongong committed:
Update javadoc

lihongong committed:
Add Java assertions to task files, Parser.java, AddCommand.java

Java assert is not currently used to ensure correct behavior.

Add Java assertions to verify preconditions and invariants across
these classes.

This improves reliability during development and testing by catching
logical errors early.

No functional changes are made. Java assert is used for debugging
purposes.

lihongong committed:
Refactor code to improve quality and maintainability

The current codebase contains inconsistent formatting and some methods
are overly complex.

Refactor multiple files to improve code quality by following best
practices.

These changes align with the Code Quality principles in CS2103T.

lihongong committed:
Refactor FindCommand.java to use Streams for generating response

The FindCommand used a for loop to iterate over the tasks and build the
response string.

The for loop is replaced with Stream operations.

Stream operations provide a more functional and readable approach to
handle the task iteration.

lihongong committed:
Merge pull request #2 from wswddl/branch-A-Assertions

Add Java assertions to task files, Parser.java, AddCommand.java

lihongong committed:
Merge branch 'master' into branch-A-CodeQuality

lihongong committed:
Merge branch 'master' into branch-A-Streams

lihongong committed:
Merge pull request #3 from wswddl/branch-A-CodeQuality

Refactor code to improve quality and maintainability

lihongong committed:
Merge branch 'master' into branch-A-Streams

# Conflicts:
#	src/main/java/milo/command/ListCommand.java

lihongong committed:
Merge pull request #4 from wswddl/branch-A-Streams

Branch a streams

lihongong committed:
Add ChangeDeadlineCommand.java RescheduleEventCommand.java

Implement B-Snooze extension where user can change the deadlines or
schedule of events.

lihongong committed:
Merge branch 'branch-B-Snooze'

lihongong committed:
Update Milo Ice messages for better user experience

Milo Ice uses certain phrases in error and normal messages that is less
user-friendly.

Improving the wording of messages enhances clarity, provides a better
user experience and makes interactions feel more natural.

Revise and refine the chatbot’s error messages and normal responses to
be clearer, friendlier and more informative.

Ensuring messages are well-structured and intuitive helps users
understand issues.

This update does not affect the chatbot's logic or functionality,
only the displayed text changes.

lihongong committed:
Merge branch 'branch-A-Personality'

lihongong committed:
Ensure start time is before end time in Event

The Event class currently allows invalid time ranges where the start
time is the same as or after the end time. Rescheduling also does not
follow this rule.

Add a validation check to enforce that the start time must be before
the end time. Throw an exception if the condition is violated.

This ensures event data remains consistent and prevents scheduling
errors.

lihongong committed:
Merge branch 'branch-A-MoreErrorHandling'

lihongong committed:
Update README.md for better clarity

Currently, the README.md has wrong title and lacks clear instructions
on how to use key features of Milo Ice.

Update the README.md by adding explanations, examples, and command
formats.

This helps introduce Milo Ice to new user and help them learn the key
features quickly.

lihongong committed:
Add a representative screenshot of the product to the docs folder

lihongong committed:
Fix typos in README.md

lihongong committed:
Fix some typos in README.md

lihongong committed:
Improve the GUI and remove Ui.java

Ui.java currently only prints messages to the terminal,
which is no longer necessary.

Remove Ui.java and refactor code to rely solely on the graphical
interface. Also tweak the CSS code to change the GUI appearance.

This change eliminates redundant code and ensures all user interactions
happen through the GUI.

No functionality is lost, as all relevant messages are now handled
within the GUI components.

lihongong committed:
Update javadoc

lihongong committed:
Refactor code to improve quality and maintainability

The current codebase contains inconsistent formatting and some methods
are overly complex.

Refactor multiple files to improve code quality by following best
practices.

These changes align with the Code Quality principles in CS2103T.

lihongong committed:
Update Milo Ice messages for better user experience

Milo Ice uses certain phrases in error and normal messages that is less
user-friendly.

Improving the wording of messages enhances clarity, provides a better
user experience and makes interactions feel more natural.

Revise and refine the chatbot’s error messages and normal responses to
be clearer, friendlier and more informative.

Ensuring messages are well-structured and intuitive helps users
understand issues.

This update does not affect the chatbot's logic or functionality,
only the displayed text changes.

lihongong committed:
Ensure start time is before end time in Event

The Event class currently allows invalid time ranges where the start
time is the same as or after the end time. Rescheduling also does not
follow this rule.

Add a validation check to enforce that the start time must be before
the end time. Throw an exception if the condition is violated.

This ensures event data remains consistent and prevents scheduling
errors.

lihongong committed:
Update README.md for better clarity

Currently, the README.md has wrong title and lacks clear instructions
on how to use key features of Milo Ice.

Update the README.md by adding explanations, examples, and command
formats.

This helps introduce Milo Ice to new user and help them learn the key
features quickly.

lihongong committed:
Improve the GUI and remove Ui.java

Ui.java currently only prints messages to the terminal,
which is no longer necessary.

Remove Ui.java and refactor code to rely solely on the graphical
interface. Also tweak the CSS code to change the GUI appearance.

This change eliminates redundant code and ensures all user interactions
happen through the GUI.

No functionality is lost, as all relevant messages are now handled
within the GUI components.

lihongong committed:
Correct MainWindow.fxml filename from MainWIndow.fxml to resolve JAR
loading issue

The JAR file is failing to load `MainWindow.fxml` because `FXMLLoader`
is case-sensitive and cannot find `view/MainWindow.fxml` due to a naming
mismatch.

The filename needs to be corrected because JAR resources are
case-sensitive and the incorrect casing causes runtime error.

Rename `MainWIndow.fxml` to `MainWindow.fxml` to match the expected
resource path in the code.

This is done to prevent `getResource()` from returning `null`.

No other changes were made, and the fix should resolve the issue without
affecting functionality.

lihongong committed:
Set custom window icon for Milo Ice application

The JavaFX application currently displays the default Java icon in the
window title bar.

This change sets a custom icon (icon.png) by loading it from the
/images/ directory in the resources folder.

Using a custom icon improves branding and enhances the visual appearance
of the application.

lihongong committed:
Fix incorrect response in ToggleMarkCommand.java

ToggleMarkCommand.java returns the same response for both marking and
unmarking a task, which is a wrong feedback.

This change updates the response to correctly differentiate between
marking and unmarking actions.

lihongong committed:
Rename image file and refactor main classes

The chat bot image file is named DaDuke.png, which does not align with
the chatbot's name, Milo Ice.

Rename DaDuke.png to DaMiloIce.png to reflect the correct chatbot name.
Also refactor Main.java and MainWindow.java.

These updates ensure consistency in branding and improve code quality.

lihongong committed:
Replace magic constants with named variables and add comments in
Parser.java

Magic constants like 4 and 6 are used in Parser.java for input length
checks.

Using named constants correctDeadlineInputLength and
correctEventInputLength to makes the code more self-explanatory.

lihongong committed:
Fix Event.java: Update start and end times only if validation succeeds

The start and end of event were updated before validation.

Update the start and end only if validation is successful.

This ensures that the event maintains a valid
time range and avoids unintended updates when an exception is thrown.
